filters/encrypt/filter.go

```diff
@@ -0,0 +1,554 @@
+		case fkind == reflect.Struct:
+			if err := ef.filterField(ctx, field, filterOverrides, opt...); err != nil {
+				return err
+			}
+
+		case isTaggable:
+			if err := ef.filterTaggable(ctx, taggedInterface, opt...); err != nil {
+				return fmt.Errorf("%s: %w", op, err)
+			}
+		}
+	}
+	return nil
+}
+
+// filterTaggable will filter data that implements the Taggable interface
+func (ef *Filter) filterTaggable(ctx context.Context, t Taggable, _ ...Option) error {
+	const op = "event.(Filter).filterTaggable"
+	if t == nil {
+		return fmt.Errorf("%s: missing taggable interface: %w", op, ErrInvalidParameter)
+	}
+	tags, err := t.Tags()
+	if err != nil {
+		return fmt.Errorf("%s: unable to get tags from taggable interface: %w", op, err)
+	}
+	for _, pt := range tags {
+		value, err := pointerstructure.Get(t, pt.Pointer)
+		if err != nil {
+			if errors.Is(err, pointerstructure.ErrNotFound) {
+				continue
+			} else {
+				return fmt.Errorf("%s: unable to get value using tag pointer structure (pointer == %s): %w", op, pt.Pointer, err)
+			}
+		}
+		rv := reflect.Indirect(reflect.ValueOf(value))
+		info := &tagInfo{
+			Classification: pt.Classification,
+			Operation:      pt.Filter,
+		}
+		if err = ef.filterValue(ctx, rv, info, withPointer(t, pt.Pointer)); err != nil {
+			return fmt.Errorf("%s: %w", op, err)
+		}
+	}
+	return nil
+}
+
+// filterSlice will filter a slice reflect.Value
+func (ef *Filter) filterSlice(ctx context.Context, classificationTag *tagInfo, slice reflect.Value, opt ...Option) error {
+	const op = "event.(Filter).filterSlice"
+	if classificationTag == nil {
+		return fmt.Errorf("%s: missing classification tag: %w", op, ErrInvalidParameter)
+	}
+	// check for nil value (prevent panics)
+	if slice == reflect.ValueOf(nil) {
+		return nil
+	}
+
+	if slice.Kind() == reflect.Ptr && !slice.IsNil() {
+		slice = slice.Elem()
+	}
+
+	ftype := slice.Type()
+	if ftype != reflect.TypeOf([]string{}) && ftype != reflect.TypeOf([]*string{}) && ftype != reflect.TypeOf([][]uint8{}) {
+		return fmt.Errorf("%s: slice parameter is not a []string or [][]byte: (%s): %w", op, slice.String(), ErrInvalidParameter)
+	}
+	if classificationTag.Classification == PublicClassification {
+		return nil
+	}
+	if slice.Len() == 0 {
+		return nil
+	}
+
+	for i := 0; i < slice.Len(); i++ {
+		if err := ef.filterValue(ctx, slice.Index(i), classificationTag); err != nil {
+			return fmt.Errorf("%s: %w", op, err)
+		}
+	}
+	return nil
+}
+
+// filterValue will filter a value based on it's DataClassification
+func (ef *Filter) filterValue(ctx context.Context, fv reflect.Value, classificationTag *tagInfo, opt ...Option) error {
+	const op = "event.(Filter).filterValue"
+	if classificationTag == nil {
+		return fmt.Errorf("%s: missing classification tag: %w", op, ErrInvalidParameter)
+	}
+
+	// check for nil value (prevent panics)
+	if fv == reflect.ValueOf(nil) {
+		return nil
+	}
+
+	if fv.Kind() == reflect.Ptr && fv.Elem().Kind() == reflect.String {
```

Comment:
What if it's a `*[]byte`? I realize you don't support that at all now, but I just thought about it seeing this so figured I'd bring it up.

Reply:
I think, I'll leave supporting `*[]byte` to a future PR.  ty!